go/teams/handler.go

```diff
@@ -269,3 +271,122 @@ func HandleOpenTeamAccessRequest(ctx context.Context, g *libkb.GlobalContext, ms
 		return team.ChangeMembership(ctx, req)
 	})
 }
+
+func HandleTeamSeitan(ctx context.Context, g *libkb.GlobalContext, msg keybase1.TeamSeitanMsg) (err error) {
+	ctx = libkb.WithLogTag(ctx, "CLKR")
+	defer g.CTrace(ctx, "HandleTeamSeitan", func() error { return err })()
+
+	team, err := Load(ctx, g, keybase1.LoadTeamArg{
+		ID:          msg.TeamID,
+		Public:      msg.TeamID.IsPublic(),
+		ForceRepoll: true,
+	})
+	if err != nil {
+		return err
+	}
+	var invitesToCancel []keybase1.TeamInviteID
+
+	var req keybase1.TeamChangeReq
+	req.CompletedInvites = make(map[keybase1.TeamInviteID]keybase1.UserVersionPercentForm)
+
+	for _, seitan := range msg.Seitans {
+		invite, found := team.chain().FindActiveInviteByID(seitan.InviteID)
+		if !found {
+			return libkb.NotFoundError{}
+		}
+
+		g.Log.CDebugf(ctx, "Processing Seitan acceptance for invite %s", invite.Id)
+
+		err := handleSeitanSingle(ctx, g, team, invite, seitan)
+		if err != nil {
+			g.Log.CDebugf(ctx, "Provided AKey failed to verify with error: \"%v\"; canceling invite.", err)
+			invitesToCancel = append(invitesToCancel, invite.Id)
+			continue
+		}
+
+		uv := NewUserVersion(seitan.Uid, seitan.EldestSeqno)
+		currentRole, err := team.MemberRole(ctx, uv)
+		if err != nil {
+			g.Log.CDebugf(ctx, "Failure in team.MemberRole: %v", err)
+			return err
```

Comment:
if you error out here, you lose `invitesToCancel`. Is that OK?

Reply:
this one is tricky, because team.MemberRole never returns an errors as of now, so I assumed if it ever will it's because of some critical failure/incompatibility that should make client back off entirely

Reply:
ok

Reply:
Same goes for other returns - the one that doesn't find invite id means that client has wrong (maybe old?) sigchain for some reason. The another one where it doesn't know the given would probably mean outdated client.